### ISSUE TITLE
Duration loss stops dropping at ~0.32 for custom dataset

### ISSUE THREAD

**Issue by brendanxwhitaker** [TensorSpeech/TensorFlowTTS#450]
Here's how the model sounds after 5k steps training on LJSpeech. You can more or less make out the text `there's no need to call me sir, Professor.`

https://drive.google.com/file/d/1xRl62kncYAab4oQGGW4pSv079g2MRj2b/view?usp=sharing

Now here's the model after 5k steps on the custom dataset I've put together. Same input text. The dataset is about half the size of LJSpeech. 15 hours or so. 

https://drive.google.com/file/d/1zOqFkHtHm9dC-HlSCZDro1qtpow30ai8/view?usp=sharing

Furthermore, the duration loss stops dropping at 0.32 for this new dataset, while it dips below 0.1 fairly quickly on LJSpeech. Are there any common causes for this? The transcription quality is quite high, the bitrates match, the audio quality is very high, there is very little silence before and after samples. The punctuation in the transcriptions has been stripped out. Is this necessary? I tried adding commas to the end of each transcription sentence, and this changed the duration extraction length, but not the quality of the audio.

**Comment by brendanxwhitaker** [comment]
@dathudeptrai I would appreciate your input on this if you have the time. Here is what the durations look like:
```Duration: [ 1 12 12  3 25 28 15  4 10  1 13 15  2 51 21  4 10 14  4  4  3  9 17  4
 27  6 10  2 10  0  3 17 12  3  3  5  1  0  0  0  0  0  0  0  0  0  0  0
  0  0  0  0  0  0  0  0  0  0  0  0  0  0]
Duration: [ 1  7  5  8 12  3  6 14  8  8 14 24 11  5  6  2  4  8  1  5  9  3  9  5
  2  2 14 20  4  3  7  3  3  3  4  3  7  6  4  3  3  9  4  3  6  3  2  5
  3  4  3  5  4  2  4  2  5  4  2  8  1  3  2  8 11 48  4  7  3  2  5  5
 10  4  5  2  5 21 11  3  2  4  3  4  6  2  4 28  1  4  0  0  0  0  0  0
  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0]
Duration: [ 1  6  2  3  4  4  5  7  5  4  3  2  4  2  0 16  9  0 10  3 10 10  9  0
  0  8  5  6  3  5 13  5  1  1  1  1  3  2  2 31 23  7  7  9 16  3  2  5
  7  3  3  8  1  3 13  0  3  8  3  2  3 22 11  7  2  6  6  7  5  7  2 16
  0  1  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0]
Duration: [ 2 12  8  5  7  5  3  3  5  6  3  4  4  8 12  7  2  7  5  1 13 13  3  9
 11 43 29  8  5  3  0  3  3  0  5  6  8 14 11 12 14 17 14  3  5  5  4  3
  6  5  3  7 11  6  6  3  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0]
Duration: [ 1 23  5  2  4  8  7  8  4  2  2  4  4  4  2  7  1  4 10  3  8  4  7  4
  4  6  4  5  3  8  4  5  5  3  6  3  4  6  3  3  3  3  2  1  2  3  2  6
  8  2  4  4 11 23 30 15  8  0  6  9  8  3  7  2  4  9 10 13  1  3  5  9
  3 16  7  5  7 23 52  9 14  3  5  4  1 12 12 14  4  7 60 29  3  4 26 11
  6  7  4  3  7  2  7  4  9  9  3  0  2  0  0  0  0  0  0  0  0  0  0  0
  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0]
Duration: [ 1  6  2  3  2  2 12  2  2  2  0  1  4  5 10  3  3  7  3  6 14 28 11  2
  4  4  4  1  9  7 12 15  8  9  8  4  0  2  7  5 16 10 27 21  9  9  5 30
  2  2 16 13  5  4 19 10  5  0  0  0  0  0  0  0  0  0  0  0  0  0  0  0
  0  0  0  0  0  0  0]```
My guess is that they aren't supposed to have so many trailing zeroes. What would cause this?

**Comment by dathudeptrai** [comment]
@brendanxwhitaker how did you extract duration for ur dataset, can you share some duration groundtruths here ?

**Comment by brendanxwhitaker** [comment]
I used this:

```
cd tensorflowtts && CUDA_VISIBLE_DEVICES=3 python ./examples/tacotron2/extract_duration.py \
  --rootdir ../dump_custom/train/ \
  --outdir ../dump_custom/train/durations/ \
  --checkpoint ../models/tacotron2-model-65000.h5 \
  --use-norm 1 \
  --config ./examples/tacotron2/conf/tacotron2.v1.yaml \
  --batch-size 32 \
  --win-front 3 \
  --win-back 3
```

What are groundtruth durations?

**Comment by dathudeptrai** [comment]
> I used this:
> 
> ```
> cd tensorflowtts && CUDA_VISIBLE_DEVICES=3 python ./examples/tacotron2/extract_duration.py \
>   --rootdir ../dump_custom/train/ \
>   --outdir ../dump_custom/train/durations/ \
>   --checkpoint ../models/tacotron2-model-65000.h5 \
>   --use-norm 1 \
>   --config ./examples/tacotron2/conf/tacotron2.v1.yaml \
>   --batch-size 32 \
>   --win-front 3 \
>   --win-back 3
> ```
> 
> What are groundtruth durations? How would I compute those? The audio files for the above examples all have spoken words up until the very end. There is not a bunch of silence at the end.

@brendanxwhitaker did you re-train tacotron-2 on ur dataset ?

**Comment by brendanxwhitaker** [comment]
I didn't! That must be the issue. I'll try it and let you know how it goes. Thanks!
